lib/util/http/bitbucket-server.ts

```diff
@@ -3,6 +3,7 @@ import { HttpBase, type InternalJsonUnsafeOptions } from './http';
 import type { HttpMethod, HttpOptions, HttpResponse } from './types';
 
 const MAX_LIMIT = 100;
+const MAX_PAGES = Number.MAX_SAFE_INTEGER;
```

Comment:
shouldn't we set it to a more conservative value like here
https://github.com/renovatebot/renovate/blob/6dee68227cca36214c0fcf6b17dc2c82ee9d5bf8/lib/util/http/bitbucket.ts#L6